homeassistant/components/binary_sensor/w800rf32.py

```diff
@@ -0,0 +1,140 @@
+https://home-assistant.io/components/binary_sensor.w800rf32/
+
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASSES_SCHEMA, PLATFORM_SCHEMA, BinarySensorDevice)
+from homeassistant.components.w800rf32 import (W800RF32_DEVICE, CONF_OFF_DELAY)
+from homeassistant.const import (CONF_DEVICE_CLASS, CONF_NAME,
+                                 ATTR_NAME, CONF_DEVICES)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import event as evt
+from homeassistant.util import dt as dt_util
+from homeassistant.helpers.dispatcher import (async_dispatcher_connect)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['w800rf32']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICES, default={}): {
+        cv.string: vol.Schema({
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+            vol.Optional(CONF_OFF_DELAY):
+            vol.Any(cv.time_period, cv.positive_timedelta)
+        })
+    },
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup_platform(hass, config,
+                               add_entities, discovery_info=None):
+    """Set up the Binary Sensor platform to w800rf32."""
+    binary_sensors = []
+    # device_id --> "c1 or a3" X10 device. entity (type dictionary)
+    # --> name, device_class etc
+    for device_id, entity in config[CONF_DEVICES].items():
+
+        _LOGGER.debug("Add %s w800rf32.binary_sensor (class %s)",
+                      entity[ATTR_NAME], entity.get(CONF_DEVICE_CLASS))
+
+        device = W800rf32BinarySensor(
+            device_id, entity.get(CONF_NAME), entity.get(CONF_DEVICE_CLASS),
+            entity.get(CONF_OFF_DELAY))
+
+        binary_sensors.append(device)
+
+    add_entities(binary_sensors)
+
+
+class W800rf32BinarySensor(BinarySensorDevice):
+    """A representation of a w800rf32 binary sensor."""
+
+    def __init__(self, device_id, name, device_class=None, off_delay=None):
+        """Initialize the w800rf32 sensor."""
+        self._device_id = device_id
+        self._signal = W800RF32_DEVICE.format(self._device_id)
+        self._name = name
+        self._should_fire_event = False
+        self._device_class = device_class
+        self._off_delay = off_delay
+        self._state = False
+        self._delay_listener = None
+
+    @callback
+    def _off_delay_listener(self, now):
+        """Switch device off after a delay."""
+        self._delay_listener = None
+        self.update_state(False)
+
+    @property
+    def name(self):
+        """Return the device name."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def should_fire_event(self):
+        """Return is the device must fire event."""
+        return self._should_fire_event
```

Comment:
Where is this used now?

Reply:
I can remove this, wasn't sure if we were supposed to have it by default.